libbeat/processors/add_remote_metadata/add_remote_metadata.go

```diff
@@ -0,0 +1,223 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package add_process_metadata
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common/atomic"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+const name = "add_remote_metadata"
+
+func init() {
+	// We cannot use this as a JS plugin as it is stateful and includes a Close method.
+	processors.RegisterPlugin(name, New)
+}
+
+var (
+	// ErrNoMatch is returned when the event doesn't contain any of the fields
+	// specified in match_pids.
+	ErrNoMatch = errors.New("none of the fields in match_keys found in the event")
+
+	// ErrNoData is returned when metadata for a metadata can't be collected.
+	ErrNoData = errors.New("metadata not found")
+
+	instanceID atomic.Uint32
+)
+
+type addRemoteMetadata struct {
+	config   config
+	mappings mapstr.M
+	src      MetadataGetter
+	cancel   context.CancelFunc
+	log      *logp.Logger
+}
+
+// Resulting processor implements `Close()` to release the cache resources.
```

Comment:
```suggestion
// Resulting processor implements `Close()` to release the cached resources.
```

Reply:
That is not the semantics that are intended, the comment refers to the cache resources (the allocs etc that are the infrastructure), not the contents of the cache.